### DISCUSSION TITLE
Proposition de BDD

### DISCUSSION THREAD

**Discussion by AdrienDSlone** [AdrienDSlone/slowTripPlanner#28] [2. Data]
-- MySQL Workbench Forward Engineering

SET @OLD_UNIQUE_CHECKS=@@UNIQUE_CHECKS, UNIQUE_CHECKS=0;
SET @OLD_FOREIGN_KEY_CHECKS=@@FOREIGN_KEY_CHECKS, FOREIGN_KEY_CHECKS=0;
SET @OLD_SQL_MODE=@@SQL_MODE, SQL_MODE='ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_ENGINE_SUBSTITUTION';

-- -----------------------------------------------------
-- Schema mydb
-- -----------------------------------------------------

-- -----------------------------------------------------
-- Schema mydb
-- -----------------------------------------------------
CREATE SCHEMA IF NOT EXISTS `mydb` DEFAULT CHARACTER SET utf8 ;
-- -----------------------------------------------------
-- Schema slowTripPlanner
-- -----------------------------------------------------
USE `mydb` ;

-- -----------------------------------------------------
-- Table `mydb`.`utilisateur`
-- -----------------------------------------------------
CREATE TABLE IF NOT EXISTS `mydb`.`utilisateur` (
  `id` INT NOT NULL,
  `nom` VARCHAR(45) NOT NULL,
  `prenom` VARCHAR(45) NOT NULL,
  `email` VARCHAR(45) NOT NULL,
  `motdepasse` VARCHAR(45) NOT NULL,
  `urlPhoto` VARCHAR(45) NULL,
  PRIMARY KEY (`id`))
ENGINE = InnoDB;

-- -----------------------------------------------------
-- Table `mydb`.`locomotion`
-- -----------------------------------------------------
CREATE TABLE IF NOT EXISTS `mydb`.`locomotion` (
  `id` INT NOT NULL,
  `intitule` VARCHAR(45) NULL,
  PRIMARY KEY (`id`))
ENGINE = InnoDB;

-- -----------------------------------------------------
-- Table `mydb`.`difficulte`
-- -----------------------------------------------------
CREATE TABLE IF NOT EXISTS `mydb`.`difficulte` (
  `id` INT NOT NULL,
  `intitule` VARCHAR(45) NOT NULL,
  PRIMARY KEY (`id`))
ENGINE = InnoDB;

-- -----------------------------------------------------
-- Table `mydb`.`preference`
-- -----------------------------------------------------
CREATE TABLE IF NOT EXISTS `mydb`.`preference` (
  `id` INT NOT NULL,
  `intitulé` VARCHAR(45) NOT NULL,
  PRIMARY KEY (`id`))
ENGINE = InnoDB;

-- -----------------------------------------------------
-- Table `mydb`.`parcours`
-- -----------------------------------------------------
CREATE TABLE IF NOT EXISTS `mydb`.`parcours` (
  `id` INT NOT NULL,
  `intitule` VARCHAR(45) NULL,
  `locomotion_id` INT NOT NULL,
  `difficulte_id` INT NOT NULL,
  `preference_id` INT NOT NULL,
  PRIMARY KEY (`id`, `locomotion_id`, `difficulte_id`, `preference_id`),
  INDEX `fk_parcours_locomotion1_idx` (`locomotion_id` ASC) VISIBLE,
  INDEX `fk_parcours_difficulte1_idx` (`difficulte_id` ASC) VISIBLE,
  INDEX `fk_parcours_preference1_idx` (`preference_id` ASC) VISIBLE,
  CONSTRAINT `fk_parcours_locomotion1`
    FOREIGN KEY (`locomotion_id`)
    REFERENCES `mydb`.`locomotion` (`id`)
    ON DELETE NO ACTION
    ON UPDATE NO ACTION,
  CONSTRAINT `fk_parcours_difficulte1`
    FOREIGN KEY (`difficulte_id`)
    REFERENCES `mydb`.`difficulte` (`id`)
    ON DELETE NO ACTION
    ON UPDATE NO ACTION,
  CONSTRAINT `fk_parcours_preference1`
    FOREIGN KEY (`preference_id`)
    REFERENCES `mydb`.`preference` (`id`)
    ON DELETE NO ACTION
    ON UPDATE NO ACTION)
ENGINE = InnoDB;

-- -----------------------------------------------------
-- Table `mydb`.`balise`
-- -----------------------------------------------------
CREATE TABLE IF NOT EXISTS `mydb`.`balise` (
  `id` INT NOT NULL,
  `date` DATE NOT NULL,
  `latitude` DOUBLE NOT NULL,
  `longitude` DOUBLE NOT NULL,
  `distance` INT NOT NULL,
  `positionDansParcours` INT NOT NULL,
  `isEtape` TINYINT NOT NULL,
  `intitule` VARCHAR(45) NULL,
  `heureArrivee` TIME NULL,
  `heureDepart` TIME NULL,
  `parcours_id` INT NOT NULL,
  `locomotion_id` INT NOT NULL,
  PRIMARY KEY (`id`, `parcours_id`, `locomotion_id`),
  INDEX `fk_balise_parcours1_idx` (`parcours_id` ASC) VISIBLE,
  INDEX `fk_balise_locomotion1_idx` (`locomotion_id` ASC) VISIBLE,
  CONSTRAINT `fk_balise_parcours1`
    FOREIGN KEY (`parcours_id`)
    REFERENCES `mydb`.`parcours` (`id`)
    ON DELETE NO ACTION
    ON UPDATE NO ACTION,
  CONSTRAINT `fk_balise_locomotion1`
    FOREIGN KEY (`locomotion_id`)
    REFERENCES `mydb`.`locomotion` (`id`)
    ON DELETE NO ACTION
    ON UPDATE NO ACTION)
ENGINE = InnoDB;

-- -----------------------------------------------------
-- Table `mydb`.`parcours_has_utilisateur`
-- -----------------------------------------------------
CREATE TABLE IF NOT EXISTS `mydb`.`parcours_has_utilisateur` (
  `parcours_id` INT NOT NULL,
  `utilisateur_id` INT NOT NULL,
  PRIMARY KEY (`parcours_id`, `utilisateur_id`),
  INDEX `fk_parcours_has_utilisateur_utilisateur1_idx` (`utilisateur_id` ASC) VISIBLE,
  INDEX `fk_parcours_has_utilisateur_parcours_idx` (`parcours_id` ASC) VISIBLE,
  CONSTRAINT `fk_parcours_has_utilisateur_parcours`
    FOREIGN KEY (`parcours_id`)
    REFERENCES `mydb`.`parcours` (`id`)
    ON DELETE NO ACTION
    ON UPDATE NO ACTION,
  CONSTRAINT `fk_parcours_has_utilisateur_utilisateur1`
    FOREIGN KEY (`utilisateur_id`)
    REFERENCES `mydb`.`utilisateur` (`id`)
    ON DELETE NO ACTION
    ON UPDATE NO ACTION)
ENGINE = InnoDB;

-- -----------------------------------------------------
-- Table `mydb`.`equipement`
-- -----------------------------------------------------
CREATE TABLE IF NOT EXISTS `mydb`.`equipement` (
  `id` INT NOT NULL,
  `intitule` VARCHAR(45) NOT NULL,
  `poids` FLOAT NULL,
  `utilisateur_id` INT NOT NULL,
  PRIMARY KEY (`id`),
  INDEX `fk_equipement_utilisateur1_idx` (`utilisateur_id` ASC) VISIBLE,
  CONSTRAINT `fk_equipement_utilisateur1`
    FOREIGN KEY (`utilisateur_id`)
    REFERENCES `mydb`.`utilisateur` (`id`)
    ON DELETE NO ACTION
    ON UPDATE NO ACTION)
ENGINE = InnoDB;

-- -----------------------------------------------------
-- Table `mydb`.`equipement_has_parcours`
-- -----------------------------------------------------
CREATE TABLE IF NOT EXISTS `mydb`.`equipement_has_parcours` (
  `equipement_id` INT NOT NULL,
  `parcours_id` INT NOT NULL,
  PRIMARY KEY (`equipement_id`, `parcours_id`),
  INDEX `fk_equipement_has_parcours_parcours1_idx` (`parcours_id` ASC) VISIBLE,
  INDEX `fk_equipement_has_parcours_equipement1_idx` (`equipement_id` ASC) VISIBLE,
  CONSTRAINT `fk_equipement_has_parcours_equipement1`
    FOREIGN KEY (`equipement_id`)
    REFERENCES `mydb`.`equipement` (`id`)
    ON DELETE NO ACTION
    ON UPDATE NO ACTION,
  CONSTRAINT `fk_equipement_has_parcours_parcours1`
    FOREIGN KEY (`parcours_id`)
    REFERENCES `mydb`.`parcours` (`id`)
    ON DELETE NO ACTION
    ON UPDATE NO ACTION)
ENGINE = InnoDB;

SET SQL_MODE=@OLD_SQL_MODE;
SET FOREIGN_KEY_CHECKS=@OLD_FOREIGN_KEY_CHECKS;
SET UNIQUE_CHECKS=@OLD_UNIQUE_CHECKS;

![bdd](https://user-images.githubusercontent.com/50139099/172566691-6448c59d-e2cb-4e54-9ed7-39d5b586a330.png)

**Comment by Scabiei** [comment]
Ci-dessous ma proposition de tables. Je n'ai pas regardé ton screenshot avant afin de ne pas être influencé. Il y a nénamoins pas mal de corrélations, c'est bon signe ^^

![proposition_bdd_Nicolas](https://user-images.githubusercontent.com/31312605/172716444-c1d83a0e-2b5e-4410-8957-40008c89bab2.png)

**Reply by AdrienDSlone** [reply]
Effectivement, beaucoup de similitudes ! c'est très bon signe !

Les points différents sont :
Les entitées "Difficulté" et "MoyenLocomotion" reliées à préférences dans ton schéma / les 2 sont reliées au parcours et l'entitée locomotion est reliée à l'entitée Balise dans mon schéma.
